plugins/rbac-backend/src/service/backend-plugin-ids-provider.ts

```diff
@@ -0,0 +1,61 @@
+import { findPaths } from '@backstage/cli-common';
+
+import * as lockfile from '@yarnpkg/lockfile';
+
+import fs from 'fs';
+
+// BackendPluginIDsProvider should retrieve list backend plugin from
+// the root yarn.lock.
+export class BackendPluginIDsProvider {
+  private readonly orgPrefixes = ['@backstage/', '@janus-idp/'];
+  private backendSuffixRegExp = new RegExp('-backend@.*$');
```

Comment:
I don't see how this can return a complete list of backend plugins within the whole instance. This considers only `@backstage/` or `@janus-idp/` packages and assumes the package name to be always suffixed with `-backend`.

By just a quick skim through the https://github.com/janus-idp/backstage-showcase/blob/main/packages/backend/package.json I can tell this will skip packages like `@immobiliarelabs/backstage-plugin-gitlab-backend`.

And that's just a simple example. Since this RBAC plugin is meant to work with any other plugins out there, I think you need a better mechanism for plugin identification.

In addition to that, the whole premise of parting `yarn.lock` is wrong, since:

1. Users are not required to use yarn (though it's very common)
2. More importantly this will skip/omit plugins installed via dynamic plugins.

The only way to determine if a package is used as a backend plugin is via its plugin instance registration.

@davidfestal do you know of any good solution here?

Reply:
Right now, we are considering just an approach for finding the static plugins with the plan to address dynamic plugins at a later point. 

Would parsing through the `package.json` be a better approach potentially? We may be over looking it, but we have not found a mechanism where backstage keeps a list of plugins that are registered.

Reply:
I'm not sure if a hacky solution that covers 80% cases is the way... What if we try a simplistic approach first?

```yaml
# app-config.yaml
rbac:
  features:
   - catalog
   - whatever
   ...
```

or as JS Array in the constructor? That should be sufficient for the static plugins that require manual TS configuration anyways... For dynamic plugins, we should have an API later on.

Reply:
I provided application configuration and constructor parameter also. I think it will  be enough for static plugins. Expecially, because rbac-backend plugin should be provided only for showcase.